vignettes/excel-editor.Rmd

```diff
@@ -0,0 +1,97 @@
+## Outline
+- Generate excel via command: `create_excel_dictionary()`
+- Make changes in excel (described below). Don't edit light grey text.
+- Either submit CA script directly or save script to file `apply_excel_dictionary()`
+
+## Making changes to excel
+
+(I think this part really needs pictures to be useful, but here's my attempt at the words)
+
+- **Replace variable metadata (Alias, Title, Description, Notes):** Modify columns E (Alias), G 
+  (Title), I (Description) and J (Notes) directly. Existing array variables are indicated by 
+  multiple rows with the same Alias, and they must have exactly one unique Title, Description
+  and Notes (you can either fill in the same value for each row or leave all except one blank).
+- **Change category metadata:**
+  - **Combine categories:** Replace column L (Code) to be the same value for the categories you
+    wish to combine. Like variable metadata (such as Title), the category metadata for combined
+    categories can only have one unique value. You can either repeat the metadata for each row
+    or leave all rows except one blank. All categories with the same code will be combined
+    regardless of whether they are next to each other.
+  - **Change category name, missing-ness, value and date:** Modify columns M (Name), N (Missing),
+    P (Value) and Q (Date) directly. For Missing, the letter "X" (lower or uppercase indicates
+    the value should be missing and empty indicates not missing). Column O (Selected) is only
+    relevant when creating Arrays, see below.
+  - **Change category order:** Move the entire category row up or down to change the category
+    order. (Be careful, footnote about how to cut and insert entire rows?) When categories
+    are combined, they are put in the order of the first occurrence of the category.
+- **Combine variables into array variables:** Arrays are created by having multiple variables 
+  of the same type with the same alias. You can do this by either setting column E (Alias) 
+  on the original row to be the same for multiple variables (in which case the original variables
+  will be automatically moved to the Hidden folder) or by copying the rows of the subvariables
+  to new rows and changing the alias on these new rows (which allows preserving the original 
+  variables outside the hidden folder). Because the original variables are always preserved
+  (at least in the hidden folder) the new array variable must have a different alias than all 
+  of the subvariables. In addition to changing the Alias, each subvariable
+  must have a unique Subvariable Label (column H) and there can be at most one 
+  Title/Description/Notes for the new array, so the values in G, I, and J may need to be updated
+  so that there's only a single unique value. Note that it is not required to move the subvariables
+  to be next to each other, but it may make the spreadsheet more legible if you do (Be careful, 
+  footnote about how to move entire rows?).
```

Comment:
can you sketch what that footnote would say?

Reply:
Definitely needs pictures or GIF, but:

When changing the order of variables and categories, it is important to keep the whole row together and in the case of variables all rows related to the variable together, including subvariables and categories. In Excel, the easiest way to move rows is to click and drag over the numbered columns on the far left of the spreadsheet to select the relevant rows, right click and select "Cut". Then click on the number of the cell at the top of the area where you want to insert the cut cells, right click and select "Insert cut cells".